legal-api/src/legal_api/resources/v2/business/business_parties.py

```diff
@@ -44,7 +44,8 @@ def get_parties(identifier, party_id=None):
         )
 
     if party_id:
-        party_roles = EntityRole.get_entity_roles_by_party_id(legal_entity.id, party_id)
+        business_id = business.id if business.is_legal_entity else business.legal_entity_id
+        party_roles = EntityRole.get_entity_roles_by_party_id(business_id, party_id)
```

Comment:
This code doesn't work for SPs.  SPs don't use `entity_roles` table except for the completing party which is not relevant to this function.  

There are 3 scenarios that you will need to take into account - SP individual, SP LEAR DBA, SP COLIN DBA.

You will need to write code to still return what the /parties endpoint previously returned for proprietors.

All 3 of these scenarios will be a bit different both in the owning record as well as the addresses to use.

SP individual - use person LE record tied to alternate name record.  also use `mailing_address_id` and `delivery_address_id` for person LE record.

SP LEAR DBA - use owning business LE record tied to alternate name record.  Use `mailing_address_id` and `delivery_address_id` from the alternate name record.

SP COLIN DBA - use owning business colin entity record tied to alternate name record.  Use `mailing_address_id` and `delivery_address_id` from the colin entity record.

Reply:
Actually, don't worry about the address bits here for now.  I forgot we have a ticket(#19667) specifically to address work.

Just update things such that a proprietor is at least returned properly for SPs when the GET parties endpoint is called.

Reply:
so this code is still a problem.  

Try the following endpoint against in dev and you should see what you should expect to get back.

``` json
GET https://legal-api-dev.apps.silver.devops.gov.bc.ca/api/v2/businesses/FM1028594/parties
{
  "parties": [
    {
      "deliveryAddress": {
        "addressCity": "Langley",
        "addressCountry": "CA",
        "addressRegion": "BC",
        "deliveryInstructions": null,
        "id": 2649561,
        "postalCode": "V2Z 2X9",
        "streetAddress": "23434 6 Ave",
        "streetAddressAdditional": null
      },
      "mailingAddress": {
        "addressCity": "Langley",
        "addressCountry": "CA",
        "addressRegion": "BC",
        "deliveryInstructions": null,
        "id": 2649562,
        "postalCode": "V2Z 2X9",
        "streetAddress": "23434 6 Ave",
        "streetAddressAdditional": null
      },
      "officer": {
        "email": "someemail..com",
        "firstName": "JANE",
        "id": 568475,
        "lastName": "DOE",
        "partyType": "person"
      },
      "roles": [
        {
          "appointmentDate": "2023-02-02",
          "cessationDate": null,
          "roleType": "Proprietor"
        }
      ]
    }
  ]
}
```

If you test locally, I don't think it won't return the same response, as the proprietor is not linked via the entity_roles table.  

As I had mentioned in my first comment, you will need to handle the 3 SP scenarios.  Don't worry about the details around which address to return for the proprietor for now.

Reply:
As per our conversation, you can fix this in your next PR.